examples/petsc/bp4.h

```diff
@@ -86,30 +88,51 @@ CEED_QFUNCTION(Diff3)(void *ctx, const CeedInt Q,
   const CeedScalar *ug = in[0], *qd = in[1];
   CeedScalar *vg = out[0];
 
+  // Quadrature Point Loop
+  CeedPragmaOMP(simd)
   for (CeedInt i=0; i<Q; i++) {
-    // Component 1
-    const CeedScalar ug00 = ug[i+(0+0*3)*Q];
-    const CeedScalar ug01 = ug[i+(0+1*3)*Q];
-    const CeedScalar ug02 = ug[i+(0+2*3)*Q];
-    vg[i+(0+0*3)*Q] = qd[i+Q*0]*ug00 + qd[i+Q*1]*ug01 + qd[i+Q*2]*ug02;
-    vg[i+(0+1*3)*Q] = qd[i+Q*1]*ug00 + qd[i+Q*3]*ug01 + qd[i+Q*4]*ug02;
-    vg[i+(0+2*3)*Q] = qd[i+Q*2]*ug00 + qd[i+Q*4]*ug01 + qd[i+Q*5]*ug02;
+    // Read spatial derivatives of u components
+    const CeedScalar uJ[3][3]     = {{ug[i+(0+0*3)*Q],
```

Comment:
Is there a mnemonic behind `uJ`? Should we call it `du` or `dudX` or something like that?

Reply:
Since **u** is a vector in this case, I simply thought `uJ` would be the most natural notation since that represents all spatial derivatives of all components of **u**, hence a Jacobian. But I can change it, of course